lib/fizzy/leb128.hpp

```diff
@@ -19,11 +19,16 @@ std::pair<T, const uint8_t*> leb128u_decode(const uint8_t* input, const uint8_t*
         if (input == end)
             throw parser_error("Unexpected EOF");
 
-        // TODO this ignores the bits in the last byte other than the least significant one
-        // So would not reject some invalid encoding with those bits set.
         result |= static_cast<T>((static_cast<T>(*input) & 0x7F) << result_shift);
         if ((*input & 0x80) == 0)
+        {
+            constexpr auto all_ones = std::numeric_limits<T>::max();
```

Comment:
```suggestion
            constexpr auto all_ones = ~T{0};
```

Reply:
Hm, earlier you changed my `~T_unsigned{0}` to `std::numeric_limits<T_unsigned>()`
https://github.com/wasmx/fizzy/commit/28b0158c3c2655c64621029fa316f768f170f9a8

Reply:
Yes, sorry. I think I wanted to see how it looks this way. Not convinced what is best.